sql/handler.cc

```diff
@@ -7374,38 +7375,74 @@ bool Vers_parse_info::check_with_conditions(const char *table_name) const
   return false;
 }
 
-bool Vers_parse_info::check_generated_type(const char *table_name,
-                                           Alter_info *alter_info,
-                                           bool integer_fields) const
+bool Vers_parse_info::check_sys_fields(const char *table_name,
+                                       Alter_info *alter_info,
+                                       bool native) const
 {
   List_iterator<Create_field> it(alter_info->create_list);
+  vers_sys_type_t found= VERS_UNDEFINED;
+  uint found_flag= 0;
   while (Create_field *f= it++)
   {
-    if (is_start(*f) || is_end(*f))
+    vers_sys_type_t check_unit= VERS_UNDEFINED;
+    uint sys_flag= f->flags & VERS_SYSTEM_FIELD;
+
+    if (!sys_flag)
+      continue;
+
+    if (sys_flag & found_flag)
```

Comment:
Always `falste`. And BTW it's already checked in `sql_yacc.yy`.

Reply:
I know. This check is natural here. Nothing wrong with double check.